# Identifying Genes Associated with Symptom Status

## Introduction

Our pipleline will use python scripts as programming language and Linux as operating system. We will use Prokka to annotate multiple bacterial genomes from the same species, then use Usearch to identify homologous genes among the genomes. Then we use python to statistically analyze if these genes are correlated with symptom status.

## Software

### 1. Prokka: for genome annotation
https://github.com/tseemann/prokka
### 2. Usearch: for sequence analysis
https://www.drive5.com/usearch/
### 3. Packages Imported for Python

 #### a. os
 https://docs.python.org/3/library/os.html
 #### b. Argparse
 https://docs.python.org/3/library/argparse.html

 #### c. Scipy.stats
 kendall tau
 https://www.statisticshowto.com/kendalls-tau/
 #### d. Pandas
 https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html
 #### e. Biopython
 https://biopython.org/wiki/Documentation
         From Biopython:
         SeqIO
         https://biopython.org/docs/1.75/api/Bio.SeqIO.html




## Scripts Included
### 1. Generate_cluster_pres_abs.py
Python script to generate a non-existence matrix for each strain and the homologous genes that it has or does not have in each gene cluster generated by Uclust.
### 2. Statistics.py 
Python script to transposing the pres/abs matrix then come out summary reports for each symptom matrix.
## Output Files
### 1. _correlation_results.txt
Hold the cluster names and their associated correlation value for each symptom
### 2. _correlation_matrix.csv
A list of significantly correlated cluster for each symptom
### 3. ResultsTable.csv
Summary information about each symptom in a table format

## Installation



In order to run this code from your working directory, use this git command to clone this repository to your workspace:
```
git clone https://github.com/lmaskeri/Identifying-Genes-Associated-with-Symptom-Status
```
Then, change working directories in order to access all files from the cloned repo:
```
cd Identifying-Genes-Associated-with-Symptom-Status
```


We will need conda to install Prokka, please make sure you have conda on your linux.
To get more information with conda:
https://docs.conda.io/en/latest/miniconda.html#linux-installers

Make sure you are downloading the currect version matched your system!

To check conda version:
```
conda -V 
```


Create the prokka environment, add the bioconda channel, install conda:

```
conda create -n prokka-env -c conda-forge -c bioconda prokka
```

Install python packages in prokka env:

Biopython:
```conda install biopython```
Pandas:
```conda install pandas```
Scipy:
```conda install scipy```

If there further questions, go back to the package instruction part and see details.

## User Directions
After the above software installation, make sure you are in Prokka-env
Then there are two options:
### 1.Test
If you type test in the terminal, our test dataset will be called. Use test assembly files in the Test Data folder.
### 2. User
If you type user in the terminal, you will need to create an Assembly folder to store assembly files. Get _correlation_results.txt, correlation_matrix.csv and ResultsTable.csv



